learning/workflow/export/customizing_html5_shell.rst

```diff
@@ -0,0 +1,220 @@
+
+Rather than the default HTML page that comes with the export templates, it is
+also possible to use a custom HTML page. This allows drastic customization of
+the final web presentation and behavior. The path to custom HTML page is
+specified in the export options as ``Html/Custom Html Shell``.
+
+The default HTML page is available in the Godot Engine repository at
+`/mist/dist/html/default.html <https://github.com/godotengine/godot/blob/master/misc/dist/html/default.html>`_.
+
+Placeholder substitution
+------------------------
+
+When exporting the game, several placeholders in the HTML page are substituted
+by values dependening on the export:
+
++------------------------------+-----------------------------------------------+
+| Placeholder                  | substituted by                                |
++==============================+===============================================+
+| ``$GODOT_BASENAME``          | Basename of exported files without suffixes,  |
+|                              | e.g. ``game`` when exporting ``game.html``    |
++------------------------------+-----------------------------------------------+
+| ``$GODOT_DEBUG_ENABLED``     | ``true`` if debugging, ``false`` otherwise    |
++------------------------------+-----------------------------------------------+
+| ``$GODOT_HEAD_INCLUDE``      | Custom string to include just before the end  |
+|                              | of the HTML ``<head>`` element                |
++------------------------------+-----------------------------------------------+
+
+The HTML file must evaluate the JavaScript file ``$GODOT_BASENAME.js``. This
+file defines a global ``Engine`` object used to start the engine, :ref:`see
+below <doc_javascript_engine_object>` for details.
+
+The boot splash image is exported as ``$GODOT_BASENAME.png`` and can be used
+e.g. in ``<img />`` elements.
+
+``$GODOT_DEBUG_ENABLED`` can be useful to optionally display e.g. an output
+console or other debug tools.
+
+``$GODOT_HEAD_INCLUDE`` is substituted with the string specified by the export
+option ``Html/Head Include``.
+
+.. _doc_javascript_engine_object:
+
+The ``Engine`` object
+---------------------
+
+The JavaScript global object ``Engine`` is defined by ``$GODOT_BASENAME.js``
+and serves as an interface to the engine start-up process.
+
+The object itself has only two methods, ``load()`` and ``unload()``.
+
+``Engine.load(basePath)``
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Loads the engine from the passed base path.
+
+Returns a promise that resolves once the engine is loaded.
+
+``Engine.unload()``
+~~~~~~~~~~~~~~~~~~~
+
+Unloads the module to free memory. This is called automatically once the
+module is instantiated unless explicitly disabled.
+
+Starting an ``Engine`` instance
+-------------------------------
+
+The more interesting interface is accessed by instantiating ``Engine`` using
+the ``new`` operator:
+
+.. code-block:: js
+
+    var engine = new Engine();
```

Comment:
Perhaps use `let` instead of `var`, as all browsers support WebAssembly support that ES2015 feature natively.